### PR TITLE
#118 Finalize Database Schema

### PR DESCRIPTION
**Pull request by gitcow**
Database: The Movie

### CODE REVIEW

**Comment by mghaught** [comment]
As I started to look this over, I think we will want to do a pass looking at missing indexes.  I'll go ahead and make comments and we can add those either on this PR or the follow-up.

File: db/migrate/20130920225539_create_users.rb [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
+      t.string :provider
+      t.string :uid
+
+      t.string :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string :unconfirmed_email # Only if using reconfirmable
+
     end
+
```
Comment (mghaught) [inline db/migrate/20130920225539_create_users.rb:38]: For missing indexes, email seems obvious and I'm thinking provider + uid for oauth logins.

**Comment by mghaught** [comment]
Actually, I bet most indexes aren't important.  Since I don't believe any single event will ever have more than 600 proposals only tables like comments, notifications, taggings and ratings will get all that large and even then we're talking under 30k rows in most cases.  I'd want to verify that postgres will even bother with using FK indexes on tables that small.  Might be a waste on the smaller tables.

**Comment by mghaught** [comment]
👍

**Comment by gitcow** [comment]
Interesting point about Postgres not bothering with indexes on smaller datasets. I'll look around for info on that once I'm done. At a minimum, I'd want basic indexes on all FK relationships, and when we have more time I'd look at optimizing compound indexes. But if the db doesn't use them, it's kinda pointless. 

Besides FK indexes, I thought about searchable field indexes for proposals, since they get searched on so much. But I think the filtering we do now is all client-side, so that's probably a wash too.

**Comment by mghaught** [comment]
Yeah, I had the same thought about the filter fields on proposals.  Since the datasets will be smaller and this app is never under that much load the common indexes are fine to go with.  Later on we can take one of our bigger events previously and do explain plans on the suspect queries to see how the indexes are/n't used.

**Pull request merged**